principal/apis/eventstream/eventstream.go

```diff
@@ -132,7 +132,7 @@ func (s *Server) newClientConnection(ctx context.Context, timeout time.Duration)
 	}
 	c.start = time.Now()
 
-	s.clusterMgr.UpdateClusterConnectionInfo(c.agentName, v1alpha1.ConnectionStatusSuccessful, c.start)
+	s.clusterMgr.SetAgentConnectionStatus(c.agentName, v1alpha1.ConnectionStatusSuccessful, c.start)
```

Comment:
Optional, just a thought: Maybe `Subscribe`  (which calls this function) would be a better place to put the call to `SetAgentConnectionStatus`? But I don't have a strong opinion.